src/vector/transaction.rs

```diff
@@ -179,9 +179,93 @@ impl Dataset {
     }
 }
 
+/// Represents maybe a transaction on a dataset for speed reasons.
+///
+/// It can be committed by calling [`commit`](MaybeTransaction::commit), but unlike [`Transaction`] it can not be rolled back. The transaction part is to make the process speedy when possible. If the [`Dataset`] does not support transaction, it does nothing.
+///
+/// If the transaction is not explicitly committed when it is dropped, it is implicitly committed, but you will not know if it fails.
+///
+/// The transaction holds a mutable borrow on the `Dataset` that it was created from, so during the
+/// lifetime of the transaction you will need to access the dataset by dereferencing the
+/// `MaybeTransaction` through its [`Deref`] or [`DerefMut`] implementations.
+#[derive(Debug)]
+pub struct MaybeTransaction<'a> {
+    dataset: &'a mut Dataset,
+    is_transaction: bool,
+    commit_on_drop: bool,
+}
+
+impl<'a> MaybeTransaction<'a> {
+    fn new(dataset: &'a mut Dataset, is_transaction: bool) -> Self {
+        MaybeTransaction {
+            dataset,
+            is_transaction,
+            commit_on_drop: true,
+        }
+    }
+
+    pub fn is_transaction(&self) -> bool {
+        self.is_transaction
+    }
+
+    /// Commits this transaction.
+    ///
+    /// If the commit fails, will return [`OGRErr::OGRERR_FAILURE`].
+    ///
+    /// Depending on drivers, this may or may not abort layer sequential readings that are active.
+    pub fn commit(mut self) -> Result<()> {
+        if !self.is_transaction {
+            return Ok(());
+        }
+
+        let rv = unsafe { gdal_sys::GDALDatasetCommitTransaction(self.dataset.c_dataset()) };
+        self.commit_on_drop = false;
+        if rv != OGRErr::OGRERR_NONE {
+            return Err(GdalError::OgrError {
+                err: rv,
+                method_name: "GDALDatasetCommitTransaction",
+            });
+        }
+        Ok(())
+    }
+}
+
+impl<'a> Deref for MaybeTransaction<'a> {
+    type Target = Dataset;
+
+    fn deref(&self) -> &Self::Target {
+        self.dataset
+    }
+}
+
+impl<'a> DerefMut for MaybeTransaction<'a> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.dataset
+    }
+}
+
+impl Dataset {
+    pub fn maybe_start_transaction(&mut self) -> MaybeTransaction<'_> {
+        let force = 1;
```

Comment:
We should add at some point that parameter for `force`, but if you want this for better performance, you absolutely don't want to enable it.

I think it only works for FileGDB, and it's implemented by making a file system-level copy of the whole dataset, and restoring it on rollback.

Reply:
Yeah that won't help with performance, I think if we want transaction for rollback purposes vs want transaction for speed purposes, I think we should have different methods.

The `MaybeTransaction` doesn't have `rollback` method at all, that'd help at compile time to let people know they can't rollback. Since it could be working on the database directly without transaction as well.

That's why I was thinking maybe some other name would be better. The current interpretation of `MaybeTransaction` is it is maybe a transaction or not, hence it has to be committed on drop, and no rollback. To make the Transaction and non transaction database do same thing.

On that note I think we can add the force thing to normal `Transaction` or add another `ForceTransaction` that does filesystem copy ourselves when Transaction fails.